src/backend/UserDb.js

```diff
@@ -41,6 +49,30 @@ class UserDb {
         const programDbPath = Path.relative(process.cwd(), "./data/ProgramDb.db");
         fs.unlinkSync(programDbPath);
     }
+
+    async getUserCount(role = "student") {
+        const doc = await this.programDb.count({ role });
+        return doc;
+    }
+
+    async getUser(username) {
+        const doc = await this.programDb.find({ username });
+        return doc;
```

Comment:
You should only return a user if the function's name is getUser. RN you return a list. (even if it is always size 1 because of our implementation)

Reply:
Agreed. Not sure about ndb but I believe find () returns either an array or cursor. You need to get the user it of it

Reply:
Updated it to use.find for the doc, which returns the first case of the array.